research-db-models.py

```diff
@@ -1,17 +1,68 @@
 from sqlalchemy import (create_engine, Table, Column, Integer,
-                        Numeric, String)
+                        Numeric, String, Boolean, Date, ForeignKey, Float)
 from sqlalchemy.ext.declarative import declarative_base
 
 Base = declarative_base()
 
+
+
+
+engine = create_engine('sqlite:///research.db')
+
+
+
+
 class FundingSource(Base):
     __tablename__ = 'funding_sources'
-    id = Column('id', Integer, autoincrement=True, primary_key=True)
-    agency = Column('agency', String())
+    funding_id = Column('funding_id', Integer, autoincrement=True, primary_key=True)
+    funding_number = Column ('funding_number',Integer())
+    funding_source = Column ('funding_source',String())
+    funding_agency = Column ('funding_agency',String())
+    duration = Column('duration', Boolean())
+    funding_contract = Column('funding_contract', Boolean())
 
 
-engine = create_engine('sqlite:///research.db')
+class  ResearchwithFund(Base):
+    __tablename__ = 'researchwithfund'
+    research_id = Column('research_id', Integer, autoincrement=True, primary_key=True)
+    research_name_th = Column ('research_name_th',String())
+    research_name_en = Column ('research_name_en',String())
+    research_field = Column ('research_field',String())
+    research_budget_thisyear  = Column('research_budget_thisyear', Integer())
+    research_budget_throughtout = Column('research_budget_throughtout', Integer())
+    research_startdate = Column('research_startdate', Date())
```

Comment:
Use data from date table instead. Changed this field to store a foreign key of the Date table.